Unhandled error

The bot crashed, currently investigating...<br>
**Console output:**
```
events.js:173
    throw err; // Unhandled 'error' event
    ^

Error [ERR_UNHANDLED_ERROR]: Unhandled error. ([object Object])
    at Client.emit (events.js:171:17)
    at WebSocketConnection.onError (/home/ec2-user/E-D-G-E/node_modules/discord.js/src/client/websocket/WebSocketConnection.js:374:17)
    at WebSocket.onError (/home/ec2-user/E-D-G-E/node_modules/ws/lib/event-target.js:128:16)
    at WebSocket.emit (events.js:182:13)
    at _receiver.cleanup (/home/ec2-user/E-D-G-E/node_modules/ws/lib/websocket.js:211:14)
    at Receiver.cleanup (/home/ec2-user/E-D-G-E/node_modules/ws/lib/receiver.js:557:13)
    at WebSocket.finalize (/home/ec2-user/E-D-G-E/node_modules/ws/lib/websocket.js:206:20)
    at TLSSocket.emit (events.js:182:13)
    at emitErrorNT (internal/streams/destroy.js:82:8)
    at emitErrorAndCloseNT (internal/streams/destroy.js:50:3)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! e-d-g-e@1.1.0 start: `node .`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the e-d-g-e@1.1.0 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/ec2-user/.npm/_logs/2018-05-16T18_44_37_329Z-debug.log
```
[2018-05-16T18_44_37_329Z-debug.log](https://github.com/N-O-D-E-Community/E-D-G-E/files/2012955/2018-05-16T18_44_37_329Z-debug.log)

--- a/ts-src/app.ts
+++ b/ts-src/app.ts
@@ -165,10 +165,20 @@ client.on("message", msg => {
     }
 });
 // LOGIN
-client.login(config.discord.token).catch(err => {
-    winston.error("Unable to login, please check \"config.json\"");
+    try {
+        client.login(config.discord.token).catch(err => {
+            winston.error("Unable to login, please check \"config.json\"");
+            winston.error(err);
+            process.exit(-1);
+        });
+    } catch (e) {
+        winston.error(e);
+        winston.info("Caught in try-catch, discarding, let's see what happens...");
+    }
+
+client.on("error", err => {
     winston.error(err);
-    process.exit(-1);
+    winston.info("Caught in 'on error', discarding, let's see what happens...");
 });
 /* END DISCORD.JS */
 });

